tests/MongoDB.Driver.Tests/Specifications/unified-test-format/UnifiedValueMatcher.cs

```diff
@@ -0,0 +1,233 @@
+                    var expectedValue = expectedElement.Value;
+
+                    if (expectedValue.IsBsonDocument &&
+                        expectedValue.AsBsonDocument.ElementCount == 1 &&
+                        expectedValue.AsBsonDocument.GetElement(0).Name.StartsWith("$$"))
+                    {
+                        var specialOperatorDocument = expectedValue.AsBsonDocument;
+                        var operatorName = specialOperatorDocument.GetElement(0).Name;
+                        var operatorValue = specialOperatorDocument[0];
+                        switch (operatorName)
+                        {
+                            case "$$exists":
+                                if (operatorValue.AsBoolean)
+                                {
+                                    actualDocument.Names.Should().Contain(expectedName);
+                                }
+                                else
+                                {
+                                    actualDocument.Names.Should().NotContain(expectedName);
+                                }
+                                continue;
+                            case "$$type":
+                                actualDocument.Names.Should().Contain(expectedName);
+                                AssertExpectedType(actualDocument[expectedName], operatorValue);
+                                continue;
+                            case "$$matchesEntity":
+                                var resultId = operatorValue.AsString;
+                                expectedValue = _entityMap.GetResult(resultId);
+                                break;
+                            case "$$matchesHexBytes":
+                                expectedValue = operatorValue;
+                                break;
+                            case "$$unsetOrMatches":
+                                if (!actualDocument.Contains(expectedName))
+                                {
+                                    continue;
+                                }
+                                expectedValue = operatorValue;
+                                break;
+                            case "$$sessionLsid":
+                                var sessionId = operatorValue.AsString;
+                                expectedValue = _entityMap.GetSessionId(sessionId);
+                                break;
+                            default:
+                                throw new FormatException($"Unrecognized special operator: '{operatorName}'");
+                        }
+                    }
+
+                    actualDocument.Names.Should().Contain(expectedName);
+                    AssertValuesMatch(actualDocument[expectedName], expectedValue, isRoot: false);
+                }
+
+                if (!isRoot)
+                {
+                    actualDocument.Names.Should().BeSubsetOf(expectedDocument.Names);
```

Comment:
I don't understand why the actual document names should be a subset of the expected document names.

Why not equal?

If the actual document is missing expected names isn't that a test failure?